singer/src/main/java/com/pinterest/singer/writer/KafkaProducerManager.java

```diff
@@ -83,6 +85,8 @@ public static void shutdown() {
       }
     }
     result = producers.get(config);
+    // log metrics for no.of kafka producers currently in the cache
+    OpenTsdbMetricConverter.addMetric(SingerMetrics.NUM_KAFKA_PRODUCERS, producers.size());
```

Comment:
Why the metric tracking is done in the getter path not done in line 81 when the new producer is created?

Reply:
Based on discussion added this metric to both create and reset producer